massa-proto/proto/massa/api/v1/versioning.proto

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023 MASSA LABS <info@massa.net>
+
+syntax = "proto3";
+
+package massa.api.v1;
+
+option csharp_namespace = "Com.Massa.Api.V1";
+option go_package = "github.com/massalabs/massa/api/v1;v1";
+option java_multiple_files = true;
+option java_package = "com.massa.api.v1";
+option objc_class_prefix = "GRPC";
+option php_namespace = "Com\\Massa\\Api\\V1";
+option ruby_package = "Com::Massa::Api::V1";
+option swift_prefix = "GRPC";
+
+// Entry for GetMipStatusResponse
+message MipStatusEntry {
+  // Mip info
+  MipInfo mip_info = 1;
+  // state id
+  ComponentStateId state = 2;
+}
+
+// Same as MipInfo struct in versioning package
+message MipInfo {
+  string name = 1;
+  uint32 version = 2;
+  uint64 start = 3;
+  uint64 timeout = 4;
+  uint64 activation_delay = 5;
+  repeated MipComponentEntry components = 6;
+}
+
+message MipComponentEntry {
+  uint32 kind = 1;
+  uint32 version = 2;
+}
+
+// Same as ComponentStateId enum in versioning package
+enum ComponentStateId {
```

Comment:
We have a kind of strict convention for enums in protobuf, we should Microsoft way 

https://learn.microsoft.com/en-us/dotnet/architecture/grpc-for-wcf-developers/protobuf-enums

```rust
// PLEASE, ALWAYS COPY AND PAST the Rust enum definition in the proto file
enum ComponentStateId {
  COMPONENT_STATE_ID_UNSPECIFIED = 0; // Default enum value
  COMPONENT_STATE_ID_ERROR = 1;  // Error statea  
  COMPONENT_STATE_ID_DEFINED = 2; // Defined state
  COMPONENT_STATE_ID_STARTED = 3; // Started state
  COMPONENT_STATE_ID_LOCKED_IN = 4; // LockedIn state
  COMPONENT_STATE_ID_ACTIVE = 5; // Active state
  COMPONENT_STATE_ID_FAILED = 6; // Failed state
}
```

Reply:
Done + add a mapping func for original enum to grpc enum